File: 자스베이스/이분탐색/upper_bound.js
const upper_bound = (nums, M) => {
  let start = 0;
  let end = nums.length - 1;
  let result = 0;
  while (start <= end) {
    let mid = parseInt((start + end) / 2);
    let temp = nums[mid];
    if (temp <= M) {
      start = mid + 1;
    } else {
      result = mid;
      end = mid - 1;
    }
  }
  return result;
};
console.log(upper_bound([1, 2, 3, 4, 4, 4, 5, 6, 7, 8], 4));

File: JavaBase/이분탐색/UpperBound.java
public class UpperBound {
    public static void main(String[] args) {
        int[] nums = { 1, 2, 3, 4, 4, 4, 5, 6, 7, 8 };
        int start = 0;
        int end = nums.length - 1;
        int M = 4;
        int result = 0;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (M < nums[mid]) {
                result = mid;
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        System.out.print(result);
    }
}
